internal/statemachine/image_definition.go

```diff
@@ -37,26 +37,23 @@ type GadgetType struct {
 
 // RootfsType defines the rootfs section of the image definition file
 type RootfsType struct {
-	AptConfig    *AptConfigType `yaml:"apt-config"    json:"AptConfig,omitempty"`
-	Seed         *SeedType      `yaml:"seed"          json:"Seed,omitempty"         jsonschema:"oneof_required=Seed"`
-	Tarball      *TarballType   `yaml:"tarball"       json:"Tarball,omitempty"      jsonschema:"oneof_required=Tarball"`
-	ArchiveTasks []string       `yaml:"archive-tasks" json:"ArchiveTasks,omitempty" jsonschema:"oneof_required=ArchiveTasks"`
-}
-
-// AptConfigType defines the apt configuration to use while
-// building the rootfs
-type AptConfigType struct {
-	Components []*string `yaml:"components"    json:"Components,omitempty"`
-	Archive    string    `yaml:"archive"       json:"Archive"                default:"ubuntu"`
-	Pocket     string    `yaml:"pocket"        json:"Pocket"                 default:"release"`
+	Components   []string     `yaml:"components"    json:"Components,omitempty"`
+	Archive      string       `yaml:"archive"       json:"Archive"                default:"ubuntu"`
+	Flavor       string       `yaml:"flavor"        json:"Flavor"                 default:"ubuntu"`
+	Pocket       string       `yaml:"pocket"        json:"Pocket"                 default:"release"`
+	Mirror       string       `yaml:"mirror"        json:"Mirror"                 default:"http://archive.ubuntu.com/ubuntu/"`
+	Seed         *SeedType    `yaml:"seed"          json:"Seed,omitempty"         jsonschema:"oneof_required=Seed"`
+	Tarball      *TarballType `yaml:"tarball"       json:"Tarball,omitempty"      jsonschema:"oneof_required=Tarball"`
+	ArchiveTasks []string     `yaml:"archive-tasks" json:"ArchiveTasks,omitempty" jsonschema:"oneof_required=ArchiveTasks"`
 }
 
 // SeedType defines the seed section of rootfs, which is used to
 // build a rootfs via seed germination
 type SeedType struct {
-	SeedURL    string   `yaml:"url"    json:"SeedURL"    jsonschema:"type=string,format=uri"`
 	SeedBranch string   `yaml:"branch" json:"SeedBranch,omitempty"`
+	SeedURLs   []string `yaml:"urls"   json:"SeedURLs"             jsonschema:"type=array,format=uri"`
 	Names      []string `yaml:"names"  json:"Names"`
+	Vcs        bool     `yaml:"vcs"    json:"Vcs"                  default:"true"`
```

Comment:
What is the purpose of this yaml argument? From what I see it's simply a boolean defining whether or not the Seed is coming from a VCS url, right? This seems a bit confusing and I think might be a bit redundant. I'd recommend, personally, to remove it and simply always add `--vcs=auto` for now. I think the value of having this configurable is quite low, as `germinate` should be able to basically guess correctly what to use no problem. The number of ambiguities that we would care about is I think minimal.

Reply:
I had an original implementation that just always used `--vcs=auto`, but that seems to make the calls to `germinate` with the seed source `https://people.canonical.com/~ubuntu-archive/seeds/` fail:

```
jawn-smith@desktop:~/sandbox/germinatejawn$ /usr/bin/germinate --mirror http://archive.ubuntu.com/ubuntu/ --arch amd64 --dist kinetic --seed-source https://people.canonical.com/~ubuntu-archive/seeds/ --seed-dist ubuntu.kinetic --no-rdepends --vcs=auto
Downloading http://archive.ubuntu.com/ubuntu/dists/kinetic/main/binary-amd64/Packages.xz file ...
Decompressing http://archive.ubuntu.com/ubuntu/dists/kinetic/main/binary-amd64/Packages.xz file ...
Downloading http://archive.ubuntu.com/ubuntu/dists/kinetic/main/source/Sources.xz file ...
Decompressing http://archive.ubuntu.com/ubuntu/dists/kinetic/main/source/Sources.xz file ...
Downloading http://archive.ubuntu.com/ubuntu/dists/kinetic/main/debian-installer/binary-amd64/Packages.xz file ...
Decompressing http://archive.ubuntu.com/ubuntu/dists/kinetic/main/debian-installer/binary-amd64/Packages.xz file ...
Downloading http://archive.ubuntu.com/ubuntu/dists/kinetic/restricted/binary-amd64/Packages.xz file ...
Decompressing http://archive.ubuntu.com/ubuntu/dists/kinetic/restricted/binary-amd64/Packages.xz file ...
Downloading http://archive.ubuntu.com/ubuntu/dists/kinetic/restricted/source/Sources.xz file ...
Decompressing http://archive.ubuntu.com/ubuntu/dists/kinetic/restricted/source/Sources.xz file ...
Downloading http://archive.ubuntu.com/ubuntu/dists/kinetic/restricted/debian-installer/binary-amd64/Packages.xz file ...
Decompressing http://archive.ubuntu.com/ubuntu/dists/kinetic/restricted/debian-installer/binary-amd64/Packages.xz file ...
* Cloning branch kinetic of https://people.canonical.com/~ubuntu-archive/seeds/ubuntu/
Cloning into '/tmp/germinate-zkaubrb4/ubuntu.kinetic'...
fatal: repository 'https://people.canonical.com/~ubuntu-archive/seeds/ubuntu/' not found
* Checking out https://people.canonical.com/~ubuntu-archive/seeds/ubuntu.kinetic/
https://people.canonical.com/~ubuntu-archive/seeds/ubuntu.kinetic/ is permanently redirected to http://people.canonical.com/~ubuntu-archive/seeds/ubuntu.kinetic/
http://people.canonical.com/~ubuntu-archive/seeds/ubuntu.kinetic/ is permanently redirected to https://people.canonical.com/~ubuntu-archive/seeds/ubuntu.kinetic/
https://people.canonical.com/~ubuntu-archive/seeds/ubuntu.kinetic/ is permanently redirected to http://people.canonical.com/~ubuntu-archive/seeds/ubuntu.kinetic/
http://people.canonical.com/~ubuntu-archive/seeds/ubuntu.kinetic/ is permanently redirected to https://people.canonical.com/~ubuntu-archive/seeds/ubuntu.kinetic/
https://people.canonical.com/~ubuntu-archive/seeds/ubuntu.kinetic/ is permanently redirected to http://people.canonical.com/~ubuntu-archive/seeds/ubuntu.kinetic/
http://people.canonical.com/~ubuntu-archive/seeds/ubuntu.kinetic/ is permanently redirected to https://people.canonical.com/~ubuntu-archive/seeds/ubuntu.kinetic/
https://people.canonical.com/~ubuntu-archive/seeds/ubuntu.kinetic/ is permanently redirected to http://people.canonical.com/~ubuntu-archive/seeds/ubuntu.kinetic/
http://people.canonical.com/~ubuntu-archive/seeds/ubuntu.kinetic/ is permanently redirected to https://people.canonical.com/~ubuntu-archive/seeds/ubuntu.kinetic/
brz: ERROR: Not a branch: "https://people.canonical.com/~ubuntu-archive/seeds/ubuntu.kinetic/".
! Could not open STRUCTURE from checkout of (any of):
!   https://people.canonical.com/~ubuntu-archive/seeds/ubuntu.kinetic
```

This same command without the `--vcs` flag succeeds. @mwhudson and I had discussed this and decided "the more explicitly stated in the image definition the better", but I'm definitely open to suggestions on how to do this better.

Reply:
Okay, as we discussed on the sync meeting, you convinced me this is the way to go. And actually looking at it, this is exactly what ubuntu-cdimage does. I am not a fan of this per-se, but this is the de-facto API of germinate, so let's go with it.